src/main/java/duke/commands/Deadline.java

```diff
@@ -0,0 +1,60 @@
+package duke.commands;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Encapsulates a Deadline object that implements a Task and has a deadline.
+ *
+ * @author Owen Tan
+ * @version duke.Duke Level-9
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param description Description to be stored in Deadline.
+     * @param by Deadline.
+     */
+    public Deadline(String description, LocalDateTime by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Overloaded constructor for Deadline.
+     *
+     * @param description Description to be stored.
+     * @param isDone Boolean that represents Deadline completion status.
+     * @param by Deadline.
+     */
+    public Deadline(String description, boolean isDone, LocalDateTime by) {
+        super(description, isDone);
+        this.by = by;
+    }
+
+    /**
+     * Returns a string representation of Deadline.
+     * @return A string representation of Deadline.
+     */
```

Comment:
I am not sure whether we need an empty line between Returns... and @return. I remember in cs2030 we need that empty line. Maybe we need to double-check that.